controllers/secret_controller.go

```diff
@@ -164,33 +165,59 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		return reconcile.Result{}, nil
 	} else {
 		// userdata secret data does not match what is expected
-		// Mark nodes configured with the previous private key for deletion
+		// Handle nodes configured with the previous private key
 		nodes := &core.NodeList{}
 		err = r.client.List(ctx, nodes, client.MatchingLabels{core.LabelOSStable: "windows"})
 		if err != nil {
 			return reconcile.Result{}, errors.Wrapf(err, "error getting node list")
 		}
-		expectedPubKeyAnno := nodeconfig.CreatePubKeyHashAnnotation(keySigner.PublicKey())
-		escapedPubKeyAnnotation := strings.Replace(nodeconfig.PubKeyHashAnnotation, "/", "~1", -1)
-		patchData := fmt.Sprintf(`[{"op":"add","path":"/metadata/annotations/%s","value":""}]`, escapedPubKeyAnnotation)
 		for _, node := range nodes.Items {
-			// Only clear the annotation for Nodes associated with Machines, as the clearing of the annotation is used
-			// solely to kick off the deletion and recreation of Machines.
 			if _, present := node.Annotations[BYOHAnnotation]; present {
-				continue
-			}
+				// For BYOH nodes, patch the username annotation, encrpyting with the new private key
+				username, err := r.usernameFromNode(ctx, &node)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
```

Comment:
Reviewing the logic here -- could I get another pair of eyes:

If `usernameFromNode` finds no associated instance then that means its entry has been removed from the ConfigMap, and the instance should be deconfigured (or is currently being deconfigured, since the configmap controller can be running in parallel). Therefore, this isn't an error scenario and should `return reconcile.Result{}, nil`.

Reply:
I think we still want an err check so maybe you could instead add in the empty instance check that returns `reconcile.Result{}, nil`